cmd/allocator/main.go

```diff
@@ -104,15 +105,47 @@ func main() {
 	httpsMux := http.NewServeMux()
 	httpsMux.HandleFunc("/v1alpha1/gameserverallocation", h.postOnly(h.allocateHandler))
 
-	caCertPool, err := getCACertPool(certDir)
+	// creates a new file watcher for client certificate folder
+	watcher, _ := fsnotify.NewWatcher()
+	defer watcher.Close() // nolint: errcheck
+	if err := watcher.Add(certDir); err != nil {
+		logger.WithError(err).Fatalf("cannot watch folder %s for secret changes", certDir)
+	}
+
+	tlsCer, err := tls.LoadX509KeyPair(tlsDir+"tls.crt", tlsDir+"tls.key")
 	if err != nil {
-		logger.WithError(err).Fatal("could not get CA certs")
+		logger.WithError(err).Fatal("server TLS could not be loaded")
 	}
+	caCertPool := loadCACertPool()
```

Comment:
I haven't looked at this closely so maybe it's ok, but I'm wondering if there is a race condition setting this variable in the go routine and reading it in the tls config. Is this code thread safe?

Reply:
It should be fine. Go routine updates the variable from a file on a single thread and tls config reads it on the main thread. Do you see a potential thread safety issue?

Reply:
If the to two routines are running on different cores and nothing tells the compiler / runtime that the variable is shared, couldn't the update be cached in the first core and never become visible to the other core? 

I feel like this is why https://godoc.org/sync/atomic#Value exists.

Reply:
I ran a simplified experiment to change a pointer to a storage location inside a loop on the go routine and read on the main thread, which is similar to what we do here and it works fine. If instead of storing storage location `a := &A{}`, I store the value, e.g. `a := A{}` then the value won't change on the main thread.

```
package main

import (
        "fmt"
)

func main() {
        type A struct {
                I int
        }

        a := &A{}
        counter := 0

        go func() {
                for {
                        counter++
                        a = &A{
                                I: counter,
                        }
                }
        }()

        for {
                fmt.Println(a)
        }
}
```

Reply:
If you want to be super safe, wrap the relevant pieces in a RWMutex?

Reply:
Done